### PR TITLE
Implement Snooze feature

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205958825930564/f

### Description
See attached task description

### Steps to test this PR
https://app.asana.com/0/0/1205989766118758/f

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3869** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3869?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3869?utm_source=stack-comment).

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/state/RealVpnStateMonitor.kt [hunk 1]
```diff
@@ -111,8 +110,9 @@ class RealVpnStateMonitor @Inject constructor(
         val runningState = when (lastState?.state) {
             ENABLING -> VpnRunningState.ENABLING
             ENABLED -> VpnRunningState.ENABLED
-            DISABLED -> VpnRunningState.DISABLED
-            else -> VpnRunningState.INVALID
+            DISABLED -> VpnRunningState.DISABLED()
+            SNOOZED -> VpnRunningState.DISABLED(TimeUnit.MINUTES.toMillis(20)) // TODO - remove hardcoded time
```
Comment (karlenDimla) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/state/RealVpnStateMonitor.kt:114]: Ideally we don’t have the hardcoded snooze length here. I wanted to remove it or replace it with a boolean, but this is currently where we get the snooze length when displaying the snooze length (https://github.com/duckduckgo/Android/pull/3869/files#diff-bb4ae2356a7050452a64491c94d8891321985471603d38084bdd0361ad741e86R129)

Still thinking how to improve this.

**Review by aitorvs — COMMENTED** [review]
Reviewed but can't approve my own PR :D

**Comment by aitorvs** [comment]
Can't approve this PR because I originally created it. But LGTM!

**Pull request merged**